[ADAM-651] Hive-style partitioning of parquet files by genomic position

jpdna opened the pull request:
Fixes #651

Manually merged changes for the "hive-style" partitioning branch as a single commit on top of master.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/2578/
Test PASSed.

jpdna commented on adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala:
```diff
@@ -1861,6 +1861,41 @@ class ADAMContext(@transient val sc: SparkContext) extends Serializable with Log
+   * @param pathName The path name to load alignment records from.
+   *   Globs/directories are supported.
+   * @param regions Optional list of genomic regions to load.
+   * @param addChrPrefix Flag to add "chr" prefix to contigs
```
agreed, I don't like it either in the API.
I'll try to push any needed conversion into the application code (Mango) by having it look at sequence dictionary and see if a conversion is needed, so that by the time a ReferenceRegion makes it into ADAM code it is on the correct contig name convention for the underlying source dataset.
@heuermh - I'll plan to use the replacement logic you pointed to - thanks!

fnothaft replied:
+1 towards not including it and pushing it to user level. FYI, you can't link against `dishevelled-bio` as it is LGPL.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/2580/
Test PASSed.

fnothaft commented on adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala:
```diff
@@ -1861,6 +1861,41 @@ class ADAMContext(@transient val sc: SparkContext) extends Serializable with Log
+    val reads: AlignmentRecordRDD = ParquetUnboundAlignmentRecordRDD(sc, pathName, sd, rgd, pgs)
+
+    val datasetBoundAlignmentRecordRDD: AlignmentRecordRDD = regions match {
+      case Some(x) => DatasetBoundAlignmentRecordRDD(reads.dataset.filter(referenceRegionsToDatasetQueryString(x)), reads.sequences, reads.recordGroups, reads.processingSteps)
```
Nit: Break longlines.

fnothaft commented on adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala:
```diff
@@ -2277,6 +2341,34 @@ class ADAMContext(@transient val sc: SparkContext) extends Serializable with Log
+   * @param addChrPrefix Flag to add "chr" prefix to contigs
+   * @return Returns a VariantRDD
+   */
+
```
Extra whitespace.

fnothaft commented on adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala:
```diff
@@ -1861,6 +1861,41 @@ class ADAMContext(@transient val sc: SparkContext) extends Serializable with Log
+   * @param addChrPrefix Flag to add "chr" prefix to contigs
+   * @return Returns an AlignmentRecordRDD.
+   */
+  def loadPartitionedParquetAlignments(pathName: String, regions: Option[Iterable[ReferenceRegion]] = None): AlignmentRecordRDD = {
```
Anywhere you have `Option[Iterable[ReferenceRegion]] = None` should be `Iterable[ReferenceRegion] = Iterable.empty`.

fnothaft commented on adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala:
```diff
@@ -2244,6 +2279,35 @@ class ADAMContext(@transient val sc: SparkContext) extends Serializable with Log
+   * @param addChrPrefix Flag to add "chr" prefix to contigs
+   * @return Returns a GenotypeRDD.
+   */
+  def loadPartitionedParquetGenotypes(pathName: String, regions: Option[Iterable[ReferenceRegion]] = None): GenotypeRDD = {
```
See above comment RE: `Option[Iterable[ReferenceRegion]] = None`.

fnothaft commented on adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala:
```diff
@@ -2277,6 +2341,34 @@ class ADAMContext(@transient val sc: SparkContext) extends Serializable with Log
+   * @return Returns a VariantRDD
+   */
+
+  def loadPartitionedParquetVariants(pathName: String, regions: Option[Iterable[ReferenceRegion]] = None): VariantRDD = {
```
See above comment RE: `Option[Iterable[ReferenceRegion]] = None`.

Also, whitespace after line.

fnothaft commented on adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala:
```diff
@@ -2593,6 +2685,32 @@ class ADAMContext(@transient val sc: SparkContext) extends Serializable with Log
+   * @return Returns a FeatureRDD.
+   */
+
+  def loadPartitionedParquetFeatures(pathName: String, regions: Option[Iterable[ReferenceRegion]] = None): FeatureRDD = {
```
See above comment RE: `Option[Iterable[ReferenceRegion]] = None`.

fnothaft commented on adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala:
```diff
@@ -2625,6 +2743,30 @@ class ADAMContext(@transient val sc: SparkContext) extends Serializable with Log
+   * @param addChrPrefix Flag to add "chr" prefix to contigs
+   * @return Returns a NucleotideContigFragmentRDD
+   */
+  def loadPartitionedParquetFragments(pathName: String, regions: Option[Iterable[ReferenceRegion]] = None): NucleotideContigFragmentRDD = {
```
See above comment RE: `Option[Iterable[ReferenceRegion]] = None`.

fnothaft commented on adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala:
```diff
@@ -3046,4 +3188,50 @@ class ADAMContext(@transient val sc: SparkContext) extends Serializable with Log
+   * @return Return True if partitioned flag found, False otherwise.
+   */
+
+  def checkPartitionedParquetFlag(filePath: String): Boolean = {
```
+1

jpdna replied:
At the moment application code (Mango) needs to itself use `checkPartitionedParquetFlag` to check to see if the data is partitioned to decide which load function (partitioned or not) to call from ADAM.  Thus `checkPartitionedParquetFlag` needs to not be private as it needs to be used outside ADAM.

This is because, due to parameter differences (no optPredicate and not opProjection) in  the partitioned parquet load functions), we don't have a way that I see to create a generic load function at in ADAM  that is agnostic to the "hive partitioned" status.

heuermh replied:
Ah right, this conversation is going in circles a bit.  Could this method be renamed `isPartitioned` or something similar, starting with `is`?

jpdna replied:
sure thing

fnothaft commented on adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala:
```diff
@@ -3046,4 +3188,50 @@ class ADAMContext(@transient val sc: SparkContext) extends Serializable with Log
+   */
+
+  def checkPartitionedParquetFlag(filePath: String): Boolean = {
+    val path = new Path(filePath, "_isPartitionedByStartPos")
```
Yeah, I'd suggest using the `getFsAndFilesWithFilter` function above. Behavior should be undefined if you have a glob but not all the paths are partitioned.

fnothaft commented on adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala:
```diff
@@ -3046,4 +3188,41 @@ class ADAMContext(@transient val sc: SparkContext) extends Serializable with Log
+
+  def referenceRegionsToDatasetQueryString(regions: Iterable[ReferenceRegion], partitionSize: Int = 1000000): String = {
+
+    var regionQueryString = "(contigName=" + "\'" + regions.head.referenceName + "\' and posBin >= \'" +
```
This will throw if `regions.isEmpty`, suggest:

```
regions.map(r => {
  // logic for a single reference region goes here
}).mkString(" or " )
```

fnothaft commented on adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala:
```diff
@@ -2517,4 +2518,29 @@ abstract class AvroGenomicRDD[T <% IndexedRecord: Manifest, U <: Product, V <: A
+
+  def writePartitionedParquetFlag(filePath: String): Boolean = {
+    val path = new Path(filePath, "_isPartitionedByStartPos")
+    val fs = path.getFileSystem(toDF().sqlContext.sparkContext.hadoopConfiguration)
```
+1, should just be `rdd.context.hadoopConfiguration`

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/2586/
Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/2587/
Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/2588/
Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/2589/
Test PASSed.

jpdna commented:
I believe I have addressed the reviewer requests above, except for the following discussed below:

1) `add loadPartitionedParquetFragments that returns FragmentRDD and loadPartitionedParquetCoverage that returns CoverageRDD.`
     I agree these should be added for completeness, but I'd rather not hold up accepting the existing changes because I don't believe Mango needs those for current demo project.

2) `java friendly method`
     would like to delay for a future update

3) `make referenceRegionsToDatasetQueryString private`
       This requires a bit of work as in  the effort with Mango to optimize latency by "caching" the handle to a dataset to be re-used with multiple filters, Mango currently uses  `referenceRegionsToDatasetQueryString`
https://github.com/jpdna/mango/blob/18369e43354f0de3f6804ab7ed83b5923d001538/mango-core/src/main/scala/org/bdgenomics/mango/models/AlignmentRecordMaterialization.scala#L224
             I agree though that this is an implementation detail that doesn't need to be the the api, and we should incorporate into the API the ability to do the dataset handle caching optimization that Mango uses.
            I think we could solve this be adding another version of the partitioned parquet load  functions that takes an existing dataset as input rather than a pathname. 
```
def filterPartitionedParquetAlignmentRecordRDDbyRegions(dataset: AlignmentRecordRDD, regions: Iterable[ReferenceRegion] = Iterable.empty): AlignmentRecordRDD = {
```
I'll go ahead and start making those changes, and testing Mango with them, but I'd like to get the rest of this PR through a second pass in parallel.

OR - as we do need to get this merged sooner rather than later for Mango - what if we leave the `referenceRegionsToDatasetQueryString` as public for now but mark it as deprecated?

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/2590/
Test PASSed.

akmorrow13 commented on adam-core/src/test/scala/org/bdgenomics/adam/rdd/contig/NucleotideContigFragmentRDDSuite.scala:
```diff
@@ -857,4 +888,5 @@ class NucleotideContigFragmentRDDSuite extends ADAMFunSuite {
 
     checkSave(variantContexts)
   }
+
```
Remove line

jpdna replied:
done

akmorrow13 commented on adam-core/src/test/scala/org/bdgenomics/adam/rdd/feature/FeatureRDDSuite.scala:
```diff
@@ -925,6 +925,33 @@ class FeatureRDDSuite extends ADAMFunSuite {
+      assert(sequenceRdd.sequences.containsRefName("aSequence"))
+    }
+
+    val inputPath = testFile("small.1.bed")
```
There are a lot of asserts here. Can you comment their purpose or break them into separate tests?

jpdna replied:
done, removed intermediate step asserts which were redundant.

akmorrow13 commented on adam-core/src/test/scala/org/bdgenomics/adam/rdd/variant/GenotypeRDDSuite.scala:
```diff
@@ -128,6 +128,15 @@ class GenotypeRDDSuite extends ADAMFunSuite {
+    genotypes.saveAsPartitionedParquet(outputPath)
+    val unfilteredGenotypes = sc.loadPartitionedParquetGenotypes(outputPath)
+    assert(unfilteredGenotypes.rdd.count === 18)
+
```
remove line

jpdna replied:
done

akmorrow13 commented on adam-core/src/test/scala/org/bdgenomics/adam/rdd/variant/VariantRDDSuite.scala:
```diff
@@ -147,6 +147,19 @@ class VariantRDDSuite extends ADAMFunSuite {
+    assert(unfilteredVariants.rdd.count === 6)
+    assert(unfilteredVariants.dataset.count === 6)
+
+    val regionsVariants = sc.loadPartitionedParquetVariants(outputPath, List(ReferenceRegion("2", 19000L, 21000L), ReferenceRegion("13", 752700L, 752750L)))
```
line break

jpdna replied:
done

akmorrow13 commented:
@jpdna it may be good to update the Mango PR https://github.com/bigdatagenomics/mango/pull/344 and make sure we have all the functionality we need in Mango included in this PR.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/2592/
Test PASSed.

akmorrow13 commented:
Jenkins, retest this please.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/2611/<h2>Build result: FAILURE</span></h2>[...truncated 7 lines...] > /home/jenkins/git2/bin/git init /home/jenkins/workspace/ADAM-prb # timeout=10Fetching upstream changes from https://github.com/bigdatagenomics/adam.git > /home/jenkins/git2/bin/git --version # timeout=10 > /home/jenkins/git2/bin/git fetch --tags --progress https://github.com/bigdatagenomics/adam.git +refs/heads/*:refs/remotes/origin/* # timeout=15 > /home/jenkins/git2/bin/git config remote.origin.url https://github.com/bigdatagenomics/adam.git # timeout=10 > /home/jenkins/git2/bin/git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10 > /home/jenkins/git2/bin/git config remote.origin.url https://github.com/bigdatagenomics/adam.git # timeout=10Fetching upstream changes from https://github.com/bigdatagenomics/adam.git > /home/jenkins/git2/bin/git fetch --tags --progress https://github.com/bigdatagenomics/adam.git +refs/pull/*:refs/remotes/origin/pr/* # timeout=15 > /home/jenkins/git2/bin/git rev-parse origin/pr/1878/merge^{commit} # timeout=10 > /home/jenkins/git2/bin/git branch -a -v --no-abbrev --contains 7894e891fec667628bd93baa7fa89b895ac6efda # timeout=10Checking out Revision 7894e891fec667628bd93baa7fa89b895ac6efda (origin/pr/1878/merge) > /home/jenkins/git2/bin/git config core.sparsecheckout # timeout=10 > /home/jenkins/git2/bin/git checkout -f 7894e891fec667628bd93baa7fa89b895ac6efda > /home/jenkins/git2/bin/git rev-list 8f53bfed72e2f1c9558dd32dc2ab63cf1fdd75b5 # timeout=10Triggering ADAM-prb ? 2.6.2,2.10,2.2.1,centosTriggering ADAM-prb ? 2.6.2,2.11,2.2.1,centosTriggering ADAM-prb ? 2.7.3,2.10,2.2.1,centosTriggering ADAM-prb ? 2.7.3,2.11,2.2.1,centosADAM-prb ? 2.6.2,2.10,2.2.1,centos completed with result FAILUREADAM-prb ? 2.6.2,2.11,2.2.1,centos completed with result FAILUREADAM-prb ? 2.7.3,2.10,2.2.1,centos completed with result FAILUREADAM-prb ? 2.7.3,2.11,2.2.1,centos completed with result FAILURENotifying endpoint 'HTTP:https://webhooks.gitter.im/e/ac8bb6e9f53357bc8aa8'
Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/2621/<h2>Build result: FAILURE</span></h2>[...truncated 7 lines...] > /home/jenkins/git2/bin/git init /home/jenkins/workspace/ADAM-prb # timeout=10Fetching upstream changes from https://github.com/bigdatagenomics/adam.git > /home/jenkins/git2/bin/git --version # timeout=10 > /home/jenkins/git2/bin/git fetch --tags --progress https://github.com/bigdatagenomics/adam.git +refs/heads/*:refs/remotes/origin/* # timeout=15 > /home/jenkins/git2/bin/git config remote.origin.url https://github.com/bigdatagenomics/adam.git # timeout=10 > /home/jenkins/git2/bin/git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10 > /home/jenkins/git2/bin/git config remote.origin.url https://github.com/bigdatagenomics/adam.git # timeout=10Fetching upstream changes from https://github.com/bigdatagenomics/adam.git > /home/jenkins/git2/bin/git fetch --tags --progress https://github.com/bigdatagenomics/adam.git +refs/pull/*:refs/remotes/origin/pr/* # timeout=15 > /home/jenkins/git2/bin/git rev-parse origin/pr/1878/merge^{commit} # timeout=10 > /home/jenkins/git2/bin/git branch -a -v --no-abbrev --contains 345e43fb2b68d9313c8354f9bf8f2a7264f5b802 # timeout=10Checking out Revision 345e43fb2b68d9313c8354f9bf8f2a7264f5b802 (origin/pr/1878/merge) > /home/jenkins/git2/bin/git config core.sparsecheckout # timeout=10 > /home/jenkins/git2/bin/git checkout -f 345e43fb2b68d9313c8354f9bf8f2a7264f5b802 > /home/jenkins/git2/bin/git rev-list 7894e891fec667628bd93baa7fa89b895ac6efda # timeout=10Triggering ADAM-prb ? 2.6.2,2.10,2.2.1,centosTriggering ADAM-prb ? 2.6.2,2.11,2.2.1,centosTriggering ADAM-prb ? 2.7.3,2.10,2.2.1,centosTriggering ADAM-prb ? 2.7.3,2.11,2.2.1,centosADAM-prb ? 2.6.2,2.10,2.2.1,centos completed with result FAILUREADAM-prb ? 2.6.2,2.11,2.2.1,centos completed with result FAILUREADAM-prb ? 2.7.3,2.10,2.2.1,centos completed with result FAILUREADAM-prb ? 2.7.3,2.11,2.2.1,centos completed with result FAILURENotifying endpoint 'HTTP:https://webhooks.gitter.im/e/ac8bb6e9f53357bc8aa8'
Test FAILed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/2622/
Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/2626/
Test PASSed.

jpdna commented:
Ping for further review.

akmorrow13 commented on adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala:
```diff
@@ -2524,4 +2525,39 @@ abstract class AvroGenomicRDD[T <% IndexedRecord: Manifest, U <: Product, V <: A
+   *
+   * @param filePath Path to save the file at.
+   */
+  def writePartitionedParquetFlag(filePath: String): Boolean = {
```
should this be private?

jpdna replied:
agree, done.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/2632/
Test PASSed.

heuermh commented:
As discussed earlier, here are two alternatives for style changes for the load methods:

Always return dataset bound RDD (always-return-dataset-bound.patch.txt)
```scala
def loadPartitionedParquetAlignments(
  pathName: String,
  regions: Iterable[ReferenceRegion] = Iterable.empty): AlignmentRecordRDD = {

  require(isPartitioned(pathName), s"Input Parquet files ($pathName) are not partitioned.")

  val reads = loadParquetAlignments(pathName, optPredicate = None, optProjection = None)

  val dataset = if (regions.nonEmpty) {
    reads.dataset.filter(referenceRegionsToDatasetQueryString(regions))
  } else {
    reads.dataset
  }

  DatasetBoundAlignmentRecordRDD(dataset, reads.sequences, reads.recordGroups, reads.processingSteps)
}
```

Return unbound or dataset bound RDD (return-unbound-or-dataset-bound.patch.txt)
```scala
def loadPartitionedParquetAlignments(
  pathName: String,
  regions: Iterable[ReferenceRegion] = Iterable.empty): AlignmentRecordRDD = {

  val reads = loadParquetAlignments(pathName, optPredicate = None, optProjection = None)

  val filteredReads = if (regions.nonEmpty) {
    require(isPartitioned(pathName), s"Input Parquet files ($pathName) are not partitioned.")

    DatasetBoundAlignmentRecordRDD(
      reads.dataset.filter(referenceRegionsToDatasetQueryString(regions)),
      reads.sequences,
      reads.recordGroups,
      reads.processingSteps
    )
  } else {
    reads
  }

  filteredReads
}
```

heuermh commented:
Sorry, github isn't allowing me to upload the referred to patches, will send via email.

jpdna commented:
Thanks @heuermh!
I'm going with the first option that always returns `DatasetBoundAlignmentRecordRDD` because we cannot allow possibility of `.rdd` being called directly on a 'ParquetUnboundAlignmentRecordRDD' backed by partitioned parquet, as the partitioned parquet can only be read as a dataset, an attempt to read as an rdd will cause an error.   Happily, once you have a    `DatasetBoundAlignmentRecordRDD` then '.rdd' will work to convert it to an rdd.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/2657/
Test PASSed.

heuermh commented:
> I'm going with the first option that always returns DatasetBoundAlignmentRecordRDD because we cannot allow possibility of .rdd being called directly on a 'ParquetUnboundAlignmentRecordRDD' backed by partitioned parquet, as the partitioned parquet can only be read as a dataset, an attempt to read as an rdd will cause an error. 

We must not have good enough unit test coverage then, because both patches passed all unit tests.  :)

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/2660/<h2>Build result: FAILURE</span></h2>[...truncated 7 lines...] > /home/jenkins/git2/bin/git init /home/jenkins/workspace/ADAM-prb # timeout=10Fetching upstream changes from https://github.com/bigdatagenomics/adam.git > /home/jenkins/git2/bin/git --version # timeout=10 > /home/jenkins/git2/bin/git fetch --tags --progress https://github.com/bigdatagenomics/adam.git +refs/heads/*:refs/remotes/origin/* # timeout=15 > /home/jenkins/git2/bin/git config remote.origin.url https://github.com/bigdatagenomics/adam.git # timeout=10 > /home/jenkins/git2/bin/git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10 > /home/jenkins/git2/bin/git config remote.origin.url https://github.com/bigdatagenomics/adam.git # timeout=10Fetching upstream changes from https://github.com/bigdatagenomics/adam.git > /home/jenkins/git2/bin/git fetch --tags --progress https://github.com/bigdatagenomics/adam.git +refs/pull/*:refs/remotes/origin/pr/* # timeout=15 > /home/jenkins/git2/bin/git rev-parse origin/pr/1878/merge^{commit} # timeout=10 > /home/jenkins/git2/bin/git branch -a -v --no-abbrev --contains d22b9ff5b20068f475bb2638f17e7b617e3fbf0a # timeout=10Checking out Revision d22b9ff5b20068f475bb2638f17e7b617e3fbf0a (origin/pr/1878/merge) > /home/jenkins/git2/bin/git config core.sparsecheckout # timeout=10 > /home/jenkins/git2/bin/git checkout -f d22b9ff5b20068f475bb2638f17e7b617e3fbf0a > /home/jenkins/git2/bin/git rev-list cba6e71c35e443936cc74ce1575ad29490279953 # timeout=10Triggering ADAM-prb ? 2.6.2,2.10,2.2.1,centosTriggering ADAM-prb ? 2.6.2,2.11,2.2.1,centosTriggering ADAM-prb ? 2.7.3,2.10,2.2.1,centosTriggering ADAM-prb ? 2.7.3,2.11,2.2.1,centosADAM-prb ? 2.6.2,2.10,2.2.1,centos completed with result FAILUREADAM-prb ? 2.6.2,2.11,2.2.1,centos completed with result FAILUREADAM-prb ? 2.7.3,2.10,2.2.1,centos completed with result FAILUREADAM-prb ? 2.7.3,2.11,2.2.1,centos completed with result FAILURENotifying endpoint 'HTTP:https://webhooks.gitter.im/e/ac8bb6e9f53357bc8aa8'
Test FAILed.

jpdna commented:
FYI - I have a working branch where the filtering has been moved into a `filterByOverlappingRegions` in `GenomicDataset` as discussed.
I'll push it tomorrow.

jpdna commented:
Replaced by #1911
